### ISSUE TITLE
Build crashes with GDSPP turned on

### ISSUE THREAD

**Issue by elvisish** [dalexeev/gdscript-preprocessor#2]
I tried an executable build and it just crashes on loading a scene with the plugin enabled, the executable works if I turn the plugin off though. I've not done much building with godot yet so I need to figure out how to debug executables, but it seems strange it would do this as it should only be stripping comments?

**Comment by dalexeev** [comment]
I'm sorry about that. The plugin is quite crude and has not been tested properly. The crash can be caused because not only comments are stripped, but also `if` conditions are checked and can be stripped/unwrapped. There is an error, now the following code

```gdscript
if not OS.is_debug_build():
    var a = 1
var a = 2
```

in release builds converted to

```gdscript
var a = 1
var a = 2
```

although it would be more correct to replace it with

```gdscript
if true:
    var a = 1
var a = 2
```

To find the reason of the error, you can export to ZIP instead of PCK and see the code after conversion.

**Comment by elvisish** [comment]
> I'm sorry about that. The plugin is quite crude and has not been tested properly. The crash can be caused because not only comments are stripped, but also `if` conditions are checked and can be stripped/unwrapped. There is an error, now the following code
> 
> ```gdscript
> if not OS.is_debug_build():
>     var a = 1
> var a = 2
> ```
> 
> in release builds converted to
> 
> ```gdscript
> var a = 1
> var a = 2
> ```
> 
> although it would be more correct to replace it with
> 
> ```gdscript
> if true:
>     var a = 1
> var a = 2
> ```
> 
> To find the reason of the error, you can export to ZIP instead of PCK and see the code after conversion.

Thanks, I have a few `OS.is_debug_build()` checks that are probably breaking it!

**Comment by dalexeev** [comment]
Hello! I updated the plugin and fixed some bugs that could lead to invalid GDScript. Please test if your issue is fixed and let me know.

**Comment by dalexeev** [comment]
I'm guessing the issue has been fixed in 9ca2c673484730d72b3002ebf2bd6820f716c3c3 since I changed the way the `EditorExportPlugin` handles scripts (use `_export_file()` instead of `_customize_resource()`). I tested the plugin on a large project which didn't work before the commit and it looks like the project is working correctly now.

Please inform me if you encounter any bugs again. Note that I removed the 4.1 branch, let me know if you still need it.

**Comment by elvisish** [comment]
> I'm guessing the issue has been fixed in 9ca2c673484730d72b3002ebf2bd6820f716c3c3 since I changed the way the `EditorExportPlugin` handles scripts (use `_export_file` instead of `_customize_resource()`). I tested the plugin on a large project which didn't work before the commit and it looks like the project is working correctly now.
> 
> 
> 
> Please inform me if you encounter any bugs again. Note that I removed the 4.1 branch, let me know if you still need it.

So sorry I missed this, I was releasing my game at the time and time got away from me! I'll be updating it with some more content and I'll give it another test on 4.2 to see if there's any problems, thanks again!
